### PR TITLE
Some enhancements for 3.13.1

### PR DESCRIPTION
**Pull request by TreyRuffy**
+ Added SingleplayerEvent and LeaveSingleplayerEvent for when a player
enters or leaves a singleplayer world
+ Added button to link Spotify
+ Added option to display if you are playing Singleplayer on Discord
Rich Presence
+ Made it easier to control Discord Rich Presence
+ Added some more images for Discord Rich Presence depending on what
server a player is on
+ Added a button to help translate in languages
* Updated the language version for added buttons
* Fixed the language selection
* Fixed some deprecated methods
* Fixed some warnings
* Fixed bug with account gui
* Allowed the Easter mode to be controlled by 5zig Reborn
- Removed unneeded imports

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,9 +15,8 @@
  * You should have received a copy of the GNU General Public License
  * along with The 5zig Mod.  If not, see <http://www.gnu.org/licenses/>.
  */
-
 subprojects {
-    version = '3.13.1'
+    version = '3.13.2'
```
Comment (RoccoDev) [inline build.gradle:19]: Change to 3.13.1 (just rebase onto latest master)

**Comment by TreyRuffy** [comment]
Fixed everything you wanted

**Comment by RoccoDev** [comment]
Can you please rebase and run the following commands in every repository (after rebasing):
```
git rm --cached -r .
git reset --hard
git add .
git commit -m "Normalize line endings"
```

**Comment by TreyRuffy** [comment]
Did those and I added you to the repo just in case you need to do something on your own

**Comment by RoccoDev** [comment]
You just need to update submodules and move the lang files back to the 44 folder and I think we should be good to go.

**Comment by TreyRuffy** [comment]
Ok I think its done

File: .classpath [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (RoccoDev) [inline .classpath:1]: Delete this file with `git rm --cached .classpath`.

File: .project [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (RoccoDev) [inline .project:1]: Delete this too

File: .settings/org.eclipse.buildship.core.prefs [hunk 4]
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
Comment (RoccoDev) [inline .settings/org.eclipse.buildship.core.prefs:1]: Delete

File: .settings/org.eclipse.core.resources.prefs [hunk 5]
```diff
@@ -0,0 +1,2 @@
+eclipse.preferences.version=1
```
Comment (RoccoDev) [inline .settings/org.eclipse.core.resources.prefs:1]: Delete

File: mod/src/main/java/eu/the5zig/mod/discord/DiscordRPCManager.java [hunk 6]
```diff
@@ -28,8 +28,7 @@
 
 public class DiscordRPCManager {
 
-    public static final long CLIENT_ID = 587300876440305685L;
-    public static final String IMG_LOGO = "logo";
+    public static final long CLIENT_ID = 695047377031135323L;
```
Comment (RoccoDev) [inline mod/src/main/java/eu/the5zig/mod/discord/DiscordRPCManager.java:31]: Please change this back to our Client ID. I can add the new features from my dashboard, if you give me a list.

File: mod/src/main/java/eu/the5zig/mod/discord/The5zigRichPresence.java [hunk 7]
```diff
@@ -20,37 +20,69 @@
...
+    public static The5zigRichPresence getDefault() {
         The5zigRichPresence presence = new The5zigRichPresence();
-        presence.text = "Menu";
-        presence.state = "Menu";
+        presence.details = "Minecraft";
+        presence.state = I18n.translate("discord.main_menu");
+        presence.largeImage = "unknown_server";
+        presence.largeImageText = I18n.translate("discord.main_menu");
+        presence.smallImage = "default";
+        presence.smallImageText = "5zig Reloaded";
```
Comment (RoccoDev) [inline mod/src/main/java/eu/the5zig/mod/discord/The5zigRichPresence.java:39]: 5zig Reloaded -> 5zig Reborn

File: mod/src/main/java/eu/the5zig/mod/gui/GuiFriends.java [hunk 8]
```diff
@@ -58,8 +58,6 @@
 
 	private boolean hoverProfile;
 
-	private AlertOverlay alertDisconnected;
```
Comment (RoccoDev) [inline mod/src/main/java/eu/the5zig/mod/gui/GuiFriends.java:61]: Why is all of this being removed?

File: mod/src/main/java/eu/the5zig/mod/gui/GuiLanguage.java [hunk 9]
```diff
@@ -61,6 +66,9 @@ protected void actionPerformed(IButton button) {
 			selectLanguage(languageSlot.getSelectedId());
 			The5zigMod.getVars().displayScreen(lastScreen);
 		}
+		if (button.getId() == 101) {
+			SocialUrls.open("https://crowdin.com/project/5zig");
```
Comment (RoccoDev) [inline mod/src/main/java/eu/the5zig/mod/gui/GuiLanguage.java:70]: Change this URL to https://translate.5zigreborn.eu

File: mod/src/main/java/eu/the5zig/mod/listener/EventListener.java [hunk 10]
```diff
@@ -146,7 +155,35 @@ public void onServerConnect(String host, int port) {
...
+			presence.setLargeImageText("Octc");
+		} else if (new ServerInstanceTimolia().handleServer(host, port)) {
+			presence.setLargeImage("timolia");
+			presence.setLargeImageText("Timolia");
+		} else {
+			presence.setLargeImage("unknown_server");
+			presence.setLargeImageText(host + (port == 25565 ? "" : ":" + port));
+		}
+		presence.setSmallImage("default");
+        presence.setSmallImageText("5zig Reloaded");
```
Comment (RoccoDev) [inline mod/src/main/java/eu/the5zig/mod/listener/EventListener.java:185]: 5zig Reborn

File: mod/src/main/java/eu/the5zig/mod/listener/EventListener.java [hunk 11]
```diff
@@ -273,5 +310,23 @@ public void onInventorySetSlot(int slot, ItemStack itemStack) {
 		SignEditEvent event = fireEvent(new SignEditEvent(lines));
 		return event.getLines();
 	}
+	
+	public void onSingleplayer() {
+		The5zigRichPresence presence = new The5zigRichPresence();
+		presence.setDetails(I18n.translate("discord.playing_on"));
+		presence.setSmallImage("default");
+        presence.setSmallImageText("5zig Reloaded");
```
Comment (RoccoDev) [inline mod/src/main/java/eu/the5zig/mod/listener/EventListener.java:318]: 5zig Reborn + fix indentation

File: mod/src/main/java/eu/the5zig/mod/manager/itunes/ITunesMacDelegate.java [hunk 12]
```diff
@@ -50,7 +50,7 @@
 
 	public ITunesMacDelegate() {
 		try {
-			script = IOUtils.toString(Thread.currentThread().getContextClassLoader().getResourceAsStream("core/itunes/iTunes.applescript"));
+			script = String.valueOf(Thread.currentThread().getContextClassLoader().getResourceAsStream("core/itunes/iTunes.applescript"));
```
Comment (RoccoDev) [inline mod/src/main/java/eu/the5zig/mod/manager/itunes/ITunesMacDelegate.java:53]: Revert, String.valueOf will call InputStream#toString, which is not what we want.

File: mod/src/main/java/eu/the5zig/mod/util/IOUtil.java [hunk 13]
```diff
@@ -47,7 +47,7 @@ public static String download(String url, boolean post) throws IOException {
 			connection.setConnectTimeout(20000);
 			connection.setReadTimeout(60000);
 			inputStream = connection.getInputStream();
-			return IOUtils.toString(inputStream);
+			return String.valueOf(inputStream);
```
Comment (RoccoDev) [inline mod/src/main/java/eu/the5zig/mod/util/IOUtil.java:50]: Revert, String.valueOf will call InputStream#toString, which is not what we want.

File: mod/src/main/java/eu/the5zig/mod/util/Updater.java [hunk 14]
```diff
@@ -59,7 +58,7 @@ public void run() {
 			URL url = new URL("https://secure.5zigreborn.eu/version?mc=" + Version.MCVERSION);
 			HttpURLConnection connection = (HttpURLConnection) url.openConnection();
 			connection.addRequestProperty("User-Agent", "5zig/" + Version.VERSION);
-			Download download = The5zigMod.gson.fromJson(IOUtils.toString(connection.getInputStream()), Download.class);
+			Download download = The5zigMod.gson.fromJson(String.valueOf(connection.getInputStream()), Download.class);
```
Comment (RoccoDev) [inline mod/src/main/java/eu/the5zig/mod/util/Updater.java:61]: Revert, String.valueOf will call InputStream#toString, which is not what we want.

File: mod/src/main/java/eu/the5zig/mod/gui/GuiFriends.java (hunk 8 above)
Comment (TreyRuffy) [reply mod/src/main/java/eu/the5zig/mod/gui/GuiFriends.java:61]: This shouldn't have been

File: mod/src/main/java/eu/the5zig/mod/discord/DiscordRPCManager.java (hunk 6 above)
Comment (TreyRuffy) [reply mod/src/main/java/eu/the5zig/mod/discord/DiscordRPCManager.java]: https://discordapp.com/api/oauth2/applications/695047377031135323/assets

https://cdn.discordapp.com/app-assets/695047377031135323/<change this to the id on the page>.png to download them

**Pull request closed without merging**